test/test_arhp.rb

```diff
@@ -13,6 +13,23 @@ def teardown
     Phenix.burn!
   end
 
+  def test_process_forking_with_connections
+    # Ensure we have a connection already
+    assert_equal(true, ActiveRecord::Base.connected?)
+
+    # Verify that when we fork, the process doesn't crash
+    pid = Process.fork do
+      if ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR == 2
+        assert_equal(false, ActiveRecord::Base.connected?) # New to Rails 5.2
```

Comment:
Did this test fail before the code change was applied?

Reply:
Yep!

Reply:
I verified the fix in 3 different ways
 * Stashing the changes to `pool_proxy.rb` and running this unit test
 * Booting an application, forking a process and making a request to the DB
 * Deploying a Rails 5.2 application with these changes and running smoke tests